client/state/action-types.js

```diff
@@ -355,6 +355,9 @@ export const HOSTING_SFTP_USERS_SET = 'HOSTING_SFTP_USERS_SET';
 export const HOSTING_SFTP_PASSWORD_RESET = 'HOSTING_SFTP_PASSWORD_RESET';
 export const HOSTING_SFTP_USER_UPDATE = 'HOSTING_SFTP_USER_UPDATE';
 export const HOSTING_RESTORE_DATABASE_PASSWORD = 'HOSTING_RESTORE_DATABASE_PASSWORD';
+export const HOSTING_PHP_VERSION_REQUEST = 'HOSTING_PHP_VERSION_REQUEST';
```

Comment:
I agree with @gwwar's suggestion of just adding two actions: https://github.com/Automattic/wp-calypso/pull/38165#discussion_r354013656

```
HOSTING_PHP_VERSION_REQUEST;
HOSTING_PHP_VERSION_SET; //can be used to both trigger a post request and update client state optimistically
```

Reply:
We need three if folks don't want the optimistic stuff, which I think is okay. It ends up being an extra `HOSTING_PHP_VERSION_SET_REQUEST` that is only used by the data-layer but ignored by our app reducers.